docs/src/HOWTOs/MsgSetNew.tla

```diff
@@ -0,0 +1,75 @@
+------------------------------- MODULE MsgSetNew ------------------------------
+
+VARIABLE 
+    \* @type: [ int: Set([x: Int]), str: Set([y: Str]) ];
+    msgs, 
+    \* @type: Bool;
+    found3,
+    \* @type: Bool;
+    foundC
+
+
+Ints == {1,2,3}
+Strs == {"a","b","c"}
+
+\* no annotation required
+Messages == [ 
+              int |-> [x: Ints],
+              str |-> [y: Strs]
+            ]
+
+Init == 
+    /\ msgs = [ int |-> {}, str |-> {} ]
+    /\ found3 = FALSE
+    /\ foundC = FALSE
+
+\* @type: ([x: Int]) => Bool; 
+SendInt(m) == 
+    msgs' = [msgs EXCEPT !.int = msgs.int \cup {m}]
```

Comment:
The problem with syntax sugar here is that there is no good way, at the level of typechecking/annotations, to create an operator, the arguments `(a,b)`of which have the relation that the type of `b` depends on the value of `a`. Concretely, you can write (but not annotate) the following:
```tla
Send(type, m) == msgs' = [ msgs EXCEPT ![type] = @ \union {m} ]
```
because the type of `m` depends on the value of `type`, so the annotation would have to be something like `Send: ( Str, ?? ) => Bool`, but you can't determine ??, unless there's a unifying type for all messages (which there generally isn't)

Reply:
Yes, the syntax sugar would need to workaround the type system. There are some ways I can think of to do this, but that's out of band for this PR.